### ISSUE TITLE
Erroneous column names generated at 'predict'

### ISSUE THREAD

**Issue by JonathanBhimani-Burrows** [manujosephv/pytorch_tabular#357]
**Describe the bug**
When fitting a model for basic binary classification, predict returns column (class) names with erroneous values 

In addition, 'prediction' column uses these erroneous class names as outputs 
I believe this is due to the label encoder in the datamodule being applied on the target column, however I have tried to reproduce the bug offline and do the re-mapping myself, yet in this case the label encoder output maps to the original class names (0/1).
Note: tested this with both integer and str labels with the same results 

**To Reproduce**
``` python
data_config = DataConfig(
    target=['Age_Group_Target'], #target should always be a list.
    continuous_cols=num_cols_list,
    categorical_cols=cat_cols_list,
)

trainer_config = TrainerConfig(
    batch_size=256,
    max_epochs=100,

    early_stopping=None,
    checkpoints="valid_loss", # Save best checkpoint monitoring val_loss
    load_best=True, # After training, load the best checkpoint
    auto_lr_find = True
)

optimizer_config = OptimizerConfig()


head_config = LinearHeadConfig(
    layers="32", # No additional layer in head, just a mapping layer to output_dim
    dropout=0.1,
    initialization="kaiming",
    activation="Softmax"
).__dict__ 

model_config = CategoryEmbeddingModelConfig(
    task="classification",
    layers="64-32-32-64",  # Number of nodes in each layer
    # layers="64-32",  # Number of nodes in each layer
    activation="ReLU", # Activation between each layers
    learning_rate = 1e-3,
    head = "LinearHead", #Linear Head
    head_config = head_config, # Linear Head Config
)



tabular_model = TabularModel(
    data_config=data_config,
    model_config=model_config,
    optimizer_config=optimizer_config,
    trainer_config=trainer_config,
)

print(tabular_model.fit(train=train, validation=test))
print(tabular_model.predict(test, include_input_features=False))
```
**Expected behavior**
column names to have the original class names, followed by '_probability'

**Screenshots**
predictions:
<img width="226" alt="image" src="https://github.com/manujosephv/pytorch_tabular/assets/44709223/cc427cca-6b1a-4885-8e9e-5089f5a0a694">

targets:
<img width="160" alt="image" src="https://github.com/manujosephv/pytorch_tabular/assets/44709223/b7d68c3b-e4ba-4343-a68b-52ddca091390">

**Desktop (please complete the following information):**

- OS:Windows 10
- Browser: Edge
- Version : 120.0.2210.77

**Smartphone (please complete the following information):**

- Device: \[e.g. iPhone6\]
- OS: \[e.g. iOS8.1\]
- Browser \[e.g. stock browser, safari\]
- Version \[e.g. 22\]

**Additional context**

**Comment by manujosephv** [comment]
by any chance, is the target also part of the `num_cols_list`? To further debug, can you also have a sample data or something which enables me to run it on my own?

**Comment by JonathanBhimani-Burrows** [comment]
Great suggestion! I had originally set it as a category, and then as a numerical column, and in both instances that's what the result was
Not passing in the target column as a numerical or categorical column solved the problem. I'm not sure the documentation includes this fact, so perhaps it could be added? 
Thanks again for the support!
Closing

**Comment by manujosephv** [comment]
yes.. I don't think the documentation includes that.. I'll probably add it in some way
